test/kubernetes/e2e/test.go

```diff
@@ -314,6 +320,85 @@ func (i *TestInstallation) PreFailHandler(ctx context.Context) {
 	kubectlGetResourcesCmd := i.Actions.Kubectl().Command(ctx, "get", strings.Join(resourcesToGet, ","), "-A", "-owide")
 	_ = kubectlGetResourcesCmd.WithStdout(clusterStateFile).WithStderr(clusterStateFile).Run()
 	clusterStateFile.WriteString("\n")
+
+	podStdOut := bytes.NewBuffer(nil)
+	podStdErr := bytes.NewBuffer(nil)
+
+	// Fetch the name of the Gloo Gateway controller pod
+	getGlooPodNameCmd := i.Actions.Kubectl().Command(ctx, "get", "pod", "-n", i.Metadata.InstallNamespace,
+		"--selector", "gloo=gloo", "--output", "jsonpath='{.items[0].metadata.name}'")
+	cmdErr := getGlooPodNameCmd.WithStdout(podStdOut).WithStderr(podStdErr).Run()
+	if cmdErr != nil {
+		i.Assertions.Require.NoError(cmdErr)
+	}
+
+	// Clean up and check the output
+	glooPodName := strings.Trim(podStdOut.String(), "'")
+	if glooPodName == "" {
+		i.Assertions.Require.NoError(fmt.Errorf("failed to get the Gloo Gateway controller pod name: %s",
+			podStdErr.String()))
+	}
+
+	// Get the metrics from the Gloo Gateway controller pod and write them to a file
+	metricsFilePath := filepath.Join(failureDir, "metrics.log")
+	metricsFile, err := os.OpenFile(metricsFilePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, os.ModePerm)
+	i.Assertions.Require.NoError(err)
+
+	// Using an ephemeral debug pod fetch the metrics from the Gloo Gateway controller
+	metricsCmd := i.Actions.Kubectl().Command(ctx, "debug", "-n", i.Metadata.InstallNamespace,
+		"-it", "--image=curlimages/curl:7.83.1", glooPodName, "--",
+		"curl", "http://localhost:9091/metrics")
+	cmdErr = metricsCmd.WithStdout(metricsFile).WithStderr(metricsFile).Run()
+	if cmdErr != nil {
```

Comment:
nit: probably doesn't need the if check

Reply:
I tried it without and it was always being treated at a failure.

Reply:
I think it doesn't like something about it explicitly being a pointer, but I'm not sure and didn't dig in.